x-pack/plugins/apm/public/utils/formatters.ts

```diff
@@ -25,21 +26,42 @@ export function asMillis(value: number, withUnit = true) {
   return `${formatted}${withUnit ? ' ms' : ''}`;
 }
 
+export function asMicros(value: number, withUnit = true) {
+  const formatted = asInteger(value);
+  return `${formatted}${withUnit ? ' μs' : ''}`;
+}
+
 export function asMillisWithDefault(value?: number) {
   if (value == null) {
```

Comment:
i know this isn't part of the PR, but just wanted to point out that `value == null` will never be reached since TS only allows `number | undefined` type for `value`

Reply:
my mistake, this line doesn't use strict equality, so the == null evaluates to true for undefined values...we should probably add a linting rule to enforce strict equality

Reply:
Oh, good catch. I was going to remove `asMillisWithDefault` and was going to use the approach suggested in https://github.com/elastic/kibana/pull/24974#discussion_r230100358.

I see two approaches: either I make `value` optional or I remove `defaultValue`. What do you think?

Reply:
> my mistake, this line doesn't use strict equality, so the == null evaluates to true for undefined values

Yes, that was also the intention. So whenever the string is `null` or `undefined` we use the default.

> we should probably add a linting rule to enforce strict equality

We are already using [this rule](https://palantir.github.io/tslint/rules/triple-equals/) that disallows this:

```ts
const a = 0 == undefined;
```

AFAICT the default is to allow double equals specifically for `null` and `undefined`. Imho this makes sense as a shorthand for `myvar === undefined && myvar === null`.

Reply:
i think with TS, these checks for different kinds of 'nil' values become less necessary. Just something to consider as we migrate more toward full TS source.

Reply:
TS will definitely eliminate a lot of runtime checks. Looking forward to that!
In this case I'm not 100% sure. Most of the usages of the formatters are still in js-land, which makes this change both easier and harder. Easier because I don't get any compile errors from the current consumers; harder because I might be breaking their contract without knowing.

Let's say that the consumers are currently calling `asTime` with a value that can be `null`, `undefined` and the actual value. Then we either need to:
 - rewrite the formatters to accept null and undefined, and keep the "nil" check in place
 - rewrite the consumers so they never call the formatters with null or undefined.